File: iosApp/iosApp/Analytics/AppCheckRepository.swift
//
//  AppCheckRepository.swift
//  iosApp
//
//  Created by Brady, Kayla on 7/18/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import FirebaseAppCheck
import Foundation
import shared

class AppCheckRepository: IAppCheckRepository {
    // swiftlint:disable:next identifier_name
    func __getToken() async -> ApiResult<shared.Token> {
        do {
            let token = try await AppCheck.appCheck().token(forcingRefresh: false)
            return ApiResultOk(data: Token(token: token.token))
        } catch {
            return ApiResultError(code: nil, message: "\(error)")
        }
    }
}

File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/repositories/AppCheckRepository.kt
package com.mbta.tid.mbta_app.android.repositories

import com.mbta.tid.mbta_app.model.response.ApiResult
import com.mbta.tid.mbta_app.repositories.IAppCheckRepository
import com.mbta.tid.mbta_app.repositories.Token

class AppCheckRepository : IAppCheckRepository {
    override suspend fun getToken(): ApiResult<Token> {
        return ApiResult.Error(message = "TODO: Implement")
    }
}
